### PR TITLE
post_group_manifests: make the plugin work on no amd64 platforms

### PR DESCRIPTION
**Pull request by jcajka**
I have been hitting this error path on ppc64le while running osbs-box. I don't think that restricting this code path to intel only is necessary, although I'm not 100% sure what other side effects this change might have.

For the record I have been using this way patched atomic-reactor for some time on ppc64le in conjunction with osbs-box and seen no issues.

### CODE REVIEW

**Comment by twaugh** [comment]
The amd64 platform is "special" for manifest lists, in that if the client does not support manifest lists then it gets an image for the amd64 platform (or 404 if there is none).

This is the logic that lead to the decision in the code for group_manifests=false to only pick the amd64 platform.

However, it doesn't really make sense to have multiple worker builds when group_manifests=false because only one is used as the result. Given that, perhaps that case (`len(source) != 1`) should be considered an error?

**Comment by jcajka** [comment]
Ok, sounds right, but what should happen if there is no intel image only non intel ones?

**Comment by lcarva** [comment]
> I think the len(source) == 1 check should be instead of the self.goarch.get(platform, platform) == 'amd64' check.

Yes that makes sense. And if it's not exactly 1 raise an error?

**Comment by twaugh** [comment]
Yes, raise an error if there were multiple workers.

**Comment by rcerven** [comment]
@twaugh  @lcarva 
well what about len(source) > 0 ?
why not just tag digests for all platforms?

also that comment needs change or we should be returning something in case group manifests are disabled, as of now it returns empty dict

**Comment by jcajka** [comment]
Updated the PR with the suggested fix and changed the comment. I hope it is now correct.
Sorry for longer iteration time.

**Comment by mlangsdorf** [comment]
It's still failing the group_manifest unit test cases.

**Comment by jcajka** [comment]
Tests should be fixed now, question is if it is correct expected way.

**Comment by jcajka** [comment]
Fixed py3 incompatibility of tests.

File: atomic_reactor/plugins/post_group_manifests.py [hunk 1]
```diff
@@ -372,9 +372,9 @@ def run(self):
             else:
                 found = False
                 for platform, digest in source.items():
-                    if self.goarch.get(platform, platform) == 'amd64':
+                    if len(source) == 1:
```
Comment (lcarva) [inline atomic_reactor/plugins/post_group_manifests.py]: Can we just move this check outside the for-loop and raise an exception there?
```python
if len(source) != 1:
  raise ValueError('Failed to find any platform')
for ....
```

**Comment by lcarva** [comment]
Updated release notes draft:
> - `post_group_manifests` runs properly without grouping manifests on non-amd64 platform

**Pull request merged**